### PR TITLE
Add code generation for moves

### LINKED ISSUE #22: Code generation for moves
Currently, the code generation script parses the available moves, but it does not actually output any generated code for them.

### PR DESCRIPTION
**Pull request by A-very-Cunning-ham**
- add code generation for moves
- update Move struct and add type for move category

More fields may need to be included in the code generation in the future but this fills out the current Move struct.

closes #22

### CODE REVIEW

File: move_test.go [hunk 1]
```diff
@@ -4,15 +4,29 @@ import "testing"
...
+	if c1.String() != "Status" {
+		t.Fail()
+	}
+	if c2.String() != "Physical" {
+		t.Fail()
+	}
+	if c3.String() != "Special" {
+		t.Fail()
+	}
+}
```
Comment (dyc3) [inline move_test.go:32]:
```suggestion
func TestMoveCategoryString(t *testing.T) {
	tests := []struct {
		value MoveCategory
		want  string
	}{
		{
			value: Status,
			want:  "Status",
		},
		{
			value: Physical,
			want:  "Physical",
		},
		{
			value: Special,
			want:  "Special",
		},
	}
	for _, tt := range tests {
		t.Run(fmt.Sprintf("MoveCategory Stringer: %s", tt.want), func(t *testing.T) {
			t.Parallel()
			got := tt.value.String()
			if got != tt.want {
				t.Errorf("Move Category (%d) got %v, want %v", tt.value, got, tt.want)
			}
		})
	}
}
```

File: move.go [hunk 2]
```diff
@@ -2,17 +2,39 @@ package pokemonbattlelib
...
+)
+
+func (c Category) String() string {
+	switch c {
+	case status:
+		return "Status"
+	case physical:
+		return "Physical"
+	case special:
+		return "Special"
```
Comment (dyc3) [inline move.go:20]: (nit)
```suggestion
func (c MoveCategory) String() string {
	switch c {
	case Status:
		return "Status"
	case Physical:
		return "Physical"
	case Special:
		return "Special"
```

File: move.go [hunk 3]
```diff
@@ -2,17 +2,39 @@ package pokemonbattlelib
...
-type Move struct{
-	Name string
-	Type string
-	Category string
-	Max_PP int
+type Move struct {
+	ID       int
+	Name     string
+	Type     int // update this to elementalType when PR #27 is complete
+	Category Category
```
Comment (dyc3) [inline move.go:31]: (nit)
```suggestion
	Category MoveCategory
```

**Review by dyc3 — CHANGES_REQUESTED** [review]
Looks good, just a couple of small changes

File: move.go [hunk 4]
```diff
@@ -2,17 +2,39 @@ package pokemonbattlelib
 
 import "fmt"
 
+type Category int
+
+const (
+	status Category = iota + 1
+	physical
+	special
```
[the remark this answers is not in the document]
Comment (A-very-Cunning-ham) [reply move.go:10]: fixed the naming, but `iota +1` is needed for consistency with the data format in `moves.csv`

File: data/gen.go [hunk 5]
```diff
@@ -298,5 +301,15 @@ func main() {
...
 
-	log.Println("TODO: generate code for moves")
+	log.Println("generating code for moves")
+	output = createCodeOutput("moves_GEN.go")
+	output.WriteString("var ALL_MOVES = []Move{\n")
+	for _, p := range moves {
+
+		// fmt.Printf("%v\n", p)
+		// TODO: add more fields
+		output.WriteString(fmt.Sprintf("\t{ID: %d, Name: %q, Type: %d, Category: %d, Max_PP: %d,"+
```
[the remark this answers is not in the document]
Comment (A-very-Cunning-ham) [reply data/gen.go]: Fixed in  0540337

**Review by dyc3 — APPROVED** [review]
Fixed merge conflicts

**Pull request merged**